codePost_api/helpers.py

```diff
@@ -209,13 +252,17 @@ def upload_submission(api_key, assignment, students, files, mode=DEFAULT_UPLOAD_
     if len(existing_submissions) == 0:
 
         # CASE 1: No existing submission => create a new submission
-
-        return post_submission(
-            api_key=api_key,
-            assignment_id=assignment_id,
-            students=students,
-            files=files
-        )
+        try:
```

Comment:
Now that we're not returning the result of `post_submission` immediately, it looks like the successive `if` causes the cases to be executed in succession as opposed to representing mutually exclusive branches. For example, the `if` statement on line 270 will be evaluated even if `len(existing_submission) == 0`, which we don't want.

Reply:
Good catch here too. I think this is the same issue as previous comment, and I think this was fixed by the commit fixing Case 1+2. Essentially I just reintroduced the `return` of the result of `post_submission`.